proposals/0245-sharing-wifi-ssid-and-password.md

```diff
@@ -7,112 +7,236 @@
-wifiPassword | string | (Optional) Specify a WiFi password to connect specified WiFi access point.
-wifiSecurity | string | (Optional) Specify a security mechanism to use.
+### Mobile API changes
+
+1. The proxy can connect to WiFi using APIs provided by the platform if SSID, password and security type are known. However, these APIs need special permissions or entitlements. 
+2. App developers can choose if they want to provide these permissions or not. Hence, app developers should be able to configure OFF WiFi auto-connection feature using proxy API. 
+3. The proxy will provide an API to app developers to configure ON/OFF WiFi auto-connection. If app developers use this API to configure ON WiFi auto-connection, the proxy should request WiFi status info from SDL Core using below API. 
+4. The proxy should send `DeviceInfo.supportWiFiAutoConnect` information along with `RegisterAppInterface` request.
+5. If the mobile device supports WiFi connection APIs provided by platform and app developer configured ON WiFi auto-connect feature, only then proxy should send `DeviceInfo.supportWiFiAutoConnect` as `true`.
+6. The proxy should request WiFi status info only if supported secondary transport is WiFi, as defined in proposal [SDL-0141](https://github.com/smartdevicelink/sdl_evolution/blob/13031dfe6d17c20fdbc673b4ac0016992d765fcd/proposals/0141-multiple-transports.md#starting-secondary-transport). This information is provided with `Start Service ACK` during version negotiations. 
+7. If the proxy did not receive WiFi credentials using `GetWiFiStatusInfo` due to some error, the proxy should request credentials again using `GetWiFiStatusInfo` only after receiving `Transport Event Update` with a valid IP address. `Transport Event Update` behavior is defined in proposal [SDL-0141](https://github.com/smartdevicelink/sdl_evolution/blob/13031dfe6d17c20fdbc673b4ac0016992d765fcd/proposals/0141-multiple-transports.md).
+8. The proxy will receive `Transport Event Update` with an empty IP address when TCP transport is unavailable. The proxy will not try connecting WiF if `Transport Event Update` with an empty IP address is received. 
+9. The proxy should try connecting to WiFi only when Driver Distraction is OFF.
+10. The proxy should try connecting to WiFi as soon as possible upon receiving `OnSystemRequest` notification with `CONNECT_WIFI` request type. 
+11. Policy table should be updated to include below-mentioned RPCs. These RPCs should be updated in aseparate `functional group`, so that, if required, the user can specifically disable WiFi credentials and status sharing with their app while allowing other RPCs. OEMs, if they require, can enable the RPC message protection feature for below-mentioned Mobile APIs.
+12. SDL Core should check the policy table and user consent to share WiFi status info before sending a response for below-mentioned RPCs. 
 
-The `wifiSecurity` field should use predefined `String` to identify WiFi security type.
+```

```

Comment:
Same comment as above regarding code block formatting here.

Reply:
Added changes in [this ](https://github.com/smartdevicelink/sdl_evolution/pull/819/commits/8cecbd20a37e7db3d90e66bb3a338fabd1aa3c93)commit.